src/Service/VendorService/PressReader/PressReaderVendorService.php

```diff
@@ -0,0 +1,135 @@
+        }
+
+        // We're lazy loading the config to avoid errors from missing config values on dependency injection
+        $this->loadConfig();
+
+        $status = new VendorStatus();
+
+        $this->progressStart('Search data well for: "'.self::VENDOR_ARCHIVE_NAME.'"');
+
+        $offset = 1;
+        try {
+            do {
+                // Search the data well for material with acSource set to "comics plus".
+                [$pidArray, $more, $offset] = $this->datawell->search(self::VENDOR_ARCHIVE_NAME, $offset);
+                $this->TransformUrls($pidArray);
+
+                // The press reader CDN insert at special image saying that the content is not updated for newest news
+                // cover. See https://i.prcdn.co/img?cid=9L09&page=1&width=1200, but the size will be under 30Kb, so we have
+                // this extra test.
+                $pidArray = array_filter($pidArray, function ($url) {
+                    $header = $this->imageValidatorService->remoteImageHeader('cf-polished', $url);
+                    if (!empty($header)) {
+                        $header = reset($header);
+                        list($label, $size) = explode('=', $header);
+                        if ($size < 30000) {
+                            // Size to little set it to null.
+                            return false;
+                        }
+                    } else {
+                        // Size header not found.
+                        return false;
+                    }
+
+                    return true;
+                });
+
+                $batchSize = \count($pidArray);
+                $this->vendorCoreService->updateOrInsertMaterials($status, $pidArray, IdentifierType::PID, $this->getVendorId(), $this->withUpdatesDate, $this->withoutQueue, $batchSize);
+
+                $this->progressMessageFormatted($status);
+                $this->progressAdvance();
+
+                if ($this->limit && $status->records >= $this->limit) {
+                    $more = false;
+                }
+            } while ($more);
+
+            $this->vendorCoreService->releaseLock($this->getVendorId());
+
+            return VendorImportResultMessage::success($status);
+        } catch (\Exception $exception) {
+            return VendorImportResultMessage::error($exception->getMessage());
+        }
+    }
+
+    /**
+     * Set config from service from DB vendor object.
+     */
+    private function loadConfig(): void
+    {
+        $vendor = $this->vendorCoreService->getVendor($this->getVendorId());
+
+        // Set the service access configuration from the vendor.
+        $this->datawell->setSearchUrl($vendor->getDataServerURI());
+        $this->datawell->setUser($vendor->getDataServerUser());
+        $this->datawell->setPassword($vendor->getDataServerPassword());
+    }
+
+    /**
+     * Transform the URL from the datawell to CDN https://i.prcdn.co/img?cid={$id}&page=1&width=1200.
```

Comment:
Adding an example of the URLs from the datawell will increase readability

Reply:
Fixed